File: fitness/routine-service/application/routine.service.ts
import { Routine, RoutineEvent } from "core";
import { RoutineDto } from "./routine.dto";
import { RoutinePublisher } from "./routine.publisher";
import { RoutineRepository } from "./routine.repository";

export class RoutineService {
  constructor(
    private repository: RoutineRepository,
    private publisher: RoutinePublisher
  ) {}

  async get(uuid: string): Promise<RoutineDto> {
    return await this.repository.get(uuid);
  }

  async getAll(userId: string): Promise<RoutineDto[]> {
    return await this.repository.getAll(userId);
  }

  async getRoutine(uuid: string): Promise<Routine> {
    return await this.repository.getRoutine(uuid);
  }

  publish(events: RoutineEvent[]) {
    this.publisher.publish(events);
  }
}

File: fitness-dimension/internal/app/routine/routine_service.go
package app

import (
	"fitness-dimension/internal/core/routine"

	"github.com/google/uuid"
)

type RoutineService struct {
	Publisher  RoutinePublisher
	Repository RoutineRepository
}

func (s *RoutineService) Get(id string) (*routine.Routine, error) {
	routineId, err := uuid.Parse(id)
	if err != nil {
		return nil, err
	}

	return s.Repository.Find(routineId)
}

func (s *RoutineService) GetAll(userId string) ([]routine.Routine, error) {
	return s.Repository.GetAll(userId)
}
